src/backend/utils/misc/queryenvironment.c

```diff
@@ -1572,6 +1584,10 @@ ENRDropTempTables(QueryEnvironment *queryEnv)
 		object.objectSubId = 0;
 		object.objectId = enr->md.reliddesc;
 		add_exact_object_address(&object, objects);
+		/*
+		 * Also register the physical file to be deleted at the end of scope.
+		 */
+		ENRDelete(enr->md.reliddesc);
```

Comment:
Why not delete the file right here rather than adding it to delete list and later calling smgrDoPendingDeletes. We can write a new function to delete a file in SMGR

Reply:
Let me look into this. The original reasoning behind doing it this way is that we need to make sure to properly handle the fact that table variables ignore rollback while temp tables respect it. However, it may be possible to pass down an isCommit flag to this function to handle table variables and temp tables separately.

Reply:
I've implemented this suggestion - now the files are immediately cleaned up during this function, which has the added benefit of not needing to modify any other function signatures or anything else in the code.